datafusion/core/tests/expr_api/mod.rs

```diff
@@ -37,14 +37,14 @@ mod simplification;
 fn test_octet_length() {
     #[rustfmt::skip]
     evaluate_expr_test(
-        octet_length(col("list")),
+        octet_length(col("id")),
```

Comment:
`octet_length` does not support list type, so change to `id`.

Reply:
octet_length takes a string, but everything is supposed to be coercible to string (for some reason). the original example should probably still work, but it is a good idea to add a test with octet_length directly on a string column

Reply:
When you say "everything is supposed to be coercable to a string" from where do you mean? Is that a SQL "standard" behavior? I don't think it is necessairly the behavior of DataFusion (or postgres)

For example, postgres doesn't automatically coerce integers to strings in all cases. Here is one:

```sql
ERROR:  operator does not exist: integer ~~ unknown
LINE 1: select 123 LIKE 'foo';
                   ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
postgres=# select 123::varchar LIKE 'foo';
 ?column?
----------
 f
(1 row)
```

Reply:
> When you say "everything is supposed to be coercable to a string" from where do you mean?

i mean this https://github.com/apache/datafusion/blob/8aafa5498024e45b58ad2068a80cf5942babe55b/datafusion/expr/src/type_coercion/functions.rs#L649-L650

> Is that a SQL "standard" behavior? 

no, i don't think so.

i am NOT proposing this behavior, or saying that such behavior is warranted.
I am just relying what the code says that DF does.